### DISCUSSION TITLE
FIT price

### DISCUSSION THREAD

**Discussion by Rujo** [amberelectric/public-api#160] [General]
Where can I find the actual feed-in price paid to me when I am exporting electricity?
Is it correct that I must calculate it from the fields 'kwh' ,'cost' and the length of the period in /sites/{siteId}/usage ?

But where in the API can I get info about the forecast of the solar feed-in?   
The information is visible on the Amber web portal under LIVE PRICE.  It would be just nicer to use the API than scraping a protected website with a lot of JavaScript code. .

**Comment by madpilot** [comment]
The prices and usage  APIs will return all general and feedIn prices. There is a type attribute that you can filter on. Check out the documentation, it’s all in there. https://app.amber.com.au/developers

**Reply by madpilot** [reply]
BTW the current price endpoint is the easiest if you want forecasts. Use the next query parameter to request the forecasts

**Reply by Rujo** [reply]
Thank you for the swift reply. I am new with Amber and still a bit rusty with APIs.

I am working on an automated Dashboard for Amber using no external hardware but just a Google Spreadsheet and Google Apps Script.
I managed to read the data from the API but now I need to figure out what they are.

With your hint I can see now that if I request forecasts with  the API call '/sites/{siteId}/prices/current' I get a pile of data with some blocks 
having ["channelType": "feedIn"],
However the explanation for  ["perKwh": -6.09459] is 'Number of cents you will pay per kilowatt-hour (c/kWh) - includes GST' is confusing.

Just to be sure, is this the actual feed-in tariff for this period?

**Reply by madpilot** [reply]
Correct. Note that the feed in price sign is opposite to the app, so negative prices means we pay you.

**Reply by petebitt99** [reply]
I've read the documewntation but it's still not 100% clear.

Is spotPerKwh the FIT?

**Reply by madpilot** [reply]
No, `perKwh` attribute on a `Interval` of type `feedIn`

**Reply by petebitt99** [reply]
Ok, so I have pasted the API call response below.  That would indicate a FIT of -7.5c/kW, yet the Amber App says it is 8c/kW
 {
    "type": "ActualInterval",
    "date": "2024-03-12",
    "duration": 30,
    "startTime": "2024-03-12T01:30:01Z",
    "endTime": "2024-03-12T02:00:00Z",
    "nemTime": "2024-03-12T12:00:00+10:00",
    "perKwh": -7.52666,
    "renewables": 32.141999999999996,
    "spotPerKwh": 7.51836,
    "channelType": "feedIn",
    "spikeStatus": "none",
    "descriptor": "high"

Feel I still have something wrong here.

**Reply by petebitt99** [reply]
This is the API call I made.

https://api.amber.com.au/v1/sites/01H92N0GC70BAZHM48P2C52J3M/prices/current?next=5&previous=1&resolution=30

**Reply by madpilot** [reply]
That's correct - the app inverts the FiT, and rounds it to the nearest c. Think of it as a ledger that is relative to the grid, so the import price in this instance will be positive because it costs you money, and the export price will be negative because it makes you money.
